drivers/include/saul.h

```diff
@@ -315,10 +316,41 @@ int saul_read_notsup(const void *dev, phydat_t *dat);
  *
  * @param[in] class_id      device class ID
  *
- * @return      string representation of the device class
- * @return      NULL if class ID is not known
+ * @return          string representation of the device class
+ * @retval  NULL    class ID is not known
+ *
+ * @deprecated  Use @ref saul_class_print or @ref saul_class_write instead
+ *
+ * @warning For classic Harvard architectures a small buffer is used to store
+ *          the string, so that subsequent (or concurrent!) calls will
+ *          overwrite the output.
+ */
+const char *saul_class_to_str(uint8_t class_id);
+
+/**
+ * @brief   Prints the class string of the given class ID
+ *
+ * @param[in] class_id      ID of the device class to print
+ */
+void saul_class_print(uint8_t class_id);
+
+/**
+ * @brief   Write the string representation of the given device class to the
+ *          given buffer
+ *
+ * @param[out]  dest        destination buffer to write to
+ * @param[in]   max_size    size of the buffer at @p dest
+ * @param[in]   class_id    ID of the device class to write
+ *
+ * @return  Number of bytes written
+ * @retval  -EOVERFLOW      buffer at @p dest is too small
+ * @retval  -EINVAL         invalid unit in @p unit
+ *
+ * @warning The function will never write a terminating zero byte
+ * @note    If you pass `NULL` for @p dest, it will return the number of bytes
+ *          it would write (regardless of @p max_size)
  */
-const char *saul_class_to_str(const uint8_t class_id);
+ssize_t saul_class_write(char *dest, size_t max_size, uint8_t class_id);
```

Comment:
```suggestion
ssize_t saul_class_write(char *dest, const size_t max_size, const uint8_t class_id);
```
the `const` in front of `class_id` will allow to not remove the `const` in front of `claas_id` of `saul_class_to_str`